### PR TITLE
Fix up "ok to test" comment.

### PR DESCRIPTION
**Pull request by spxtr**
This change is [Reviewable](https://reviewable.io/reviews/kubernetes/test-infra/507)

### CODE REVIEW

**Comment by spxtr** [comment]
When the PR is opened, I don't need to look for issue comments and the like in order to check if I should build or ask "ok to test". This fixes a rare flake where GitHub sends us the PR opened hook but when we try to list issue comments it fails because the API hasn't learned about the new PR or something. It also makes it use less tokens.

There needs to be a test for this, but I'd like the fix to go in ASAP, since I've seen that flake once in a day of usage.

File: ciongke/cmd/hook/githubagent.go [hunk 1]
```diff
@@ -86,18 +86,29 @@ func (ga *GitHubAgent) handlePullRequestEvent(pr github.PullRequestEvent) error
 	log.Printf("%s/%s#%d %s: %s", owner, name, pr.Number, pr.Action, pr.PullRequest.HTMLURL)
 
 	switch pr.Action {
-	case "opened", "reopened", "synchronize":
-		trusted, err := ga.trustedPullRequest(pr.PullRequest)
+	case "opened":
+		// When a PR is opened, if the author is in the org then build it.
```
Comment (ixdy) [inline ciongke/cmd/hook/githubagent.go]: might be worth adding your PR comment here, noting why we special-case "opened".

File: ciongke/cmd/hook/githubagent.go [hunk 2]
```diff
@@ -86,18 +86,29 @@ func (ga *GitHubAgent) handlePullRequestEvent(pr github.PullRequestEvent) error
...
+		// Otherwise, ask for "ok to test".
+		member, err := ga.GitHubClient.IsMember(pr.PullRequest.User.Login, ga.Org)
```
Comment (ixdy) [inline ciongke/cmd/hook/githubagent.go]: it looks like you reversed the order of args vs. what's in trustedPullRequest

File: ciongke/cmd/hook/githubagent.go (hunk 1 above)
Comment (ixdy) [inline ciongke/cmd/hook/githubagent.go]: including the note about the PR not being synchronized and saving API tokens might be helpful

File: ciongke/cmd/hook/githubagent.go (hunk 2 above)
Comment (spxtr) [inline ciongke/cmd/hook/githubagent.go]: Good catch, that's the sort of thing I would like a unit test to catch.

Comment (ixdy) [inline ciongke/cmd/hook/githubagent.go]: AT&T vs Intel syntax :)

File: ciongke/cmd/hook/githubagent.go (hunk 1 above)
Comment (spxtr) [inline ciongke/cmd/hook/githubagent.go]: Done.

**Comment by rmmh** [comment]
We might want to extend the client to have configurable retries for 404s or 500s.

LGTM overall.

**Comment by spxtr** [comment]
Agreed. It's pretty bare-bones ATM. I'm particularly worried about the Jenkins API being flaky, but it hasn't had any trouble so far.

**Pull request merged**